src_plugins/show/ppx_deriving_show.cppo.ml

```diff
@@ -173,20 +173,31 @@ let rec expr_of_typ quoter typ =
     | { ptyp_desc = Ptyp_variant (fields, _, _); ptyp_loc } ->
       let cases =
         fields |> List.map (fun field ->
-          match field.prf_desc with
-          | Rtag(label, true (*empty*), []) ->
+          match attr_printer field.prf_attributes, field.prf_desc with
+          | None, Rtag(label, true (*empty*), []) ->
             let label = label.txt in
             Exp.case (Pat.variant label None)
                      [%expr Ppx_deriving_runtime.Format.pp_print_string fmt [%e str ("`" ^ label)]]
-          | Rtag(label, false, [typ]) ->
+          | Some printer, Rtag(label, true (*empty*), []) ->
+            let label = label.txt in
+            Exp.case (Pat.variant label None)
+                     [%expr [%e printer] fmt ()]
```

Comment:
I think it would be easier to factorize the two cases better, the code is identical except for the second argument of `Exp.case` that depends on the optional printer attribute.